### PR TITLE
Add docs for bank-account-requests

### PR DESCRIPTION
**Pull request by IsaacMurtagh**

### CODE REVIEW

File: src/api/bank-accounts/bank-account-requests.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  auth 'api-key'
+  example {
+    body ({
+      accountId: 'uooxSens6ykJaim1Cu1Q55',
+      type: 'quartz-details',
+    })
+  }
+{% endreqspec %}
+
+{% h4 Required Fields %}
```
Comment (ncjones) [inline src/api/bank-accounts/bank-account-requests.md:60]: You should join optional and required fields. See https://github.com/centrapay/centrapay-docs/blob/master/src/api/batch/verifone-terminal-status.md?plain=1#L34

File: src/api/bank-accounts/bank-account-requests.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      code: 'hKpKupTM391pE10xfQiorMxXarRKAHRhTfH_x',
+    })
+  }
+{% endreqspec %}
+
+{% h4 Required Fields %}
+
+| Field |  Type  |                      Description                      |
+| :---- | :----- | :---------------------------------------------------- |
+| code  | String | authorization code returned returned from third party |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:103]: Would be good to just clean this up by capitalising the description and adding a full stop.

File: src/api/bank-accounts/bank-account-requests.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+| status                   | String             | The current status of the Bank Account Request. Supported values are `created` and `done`. |
+| createdAt                | {% dt Timestamp %} | When the Bank Account Request was created.                                                 |
+| createdBy                | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                 |
+| updatedAt                | {% dt Timestamp %} | When the Bank Account Request was updated.                                                 |
+| updatedBy                | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                 |
+| authorizationRedirectUrl | String             | A redirect url to begin the OAuth 2 flow.                                                  |
+
+### Bank Account Request Types
+
+|     Field      |                                   description                                    |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:41]: Also missing capitalisation here on `descripion`

Comment (ncjones) [reply src/api/bank-accounts/bank-account-requests.md:39]:
```suggestion
|     Field      |                                   Description                                    |
```

File: src/api/bank-accounts/bank-account-requests.md (hunk 2 above)
Comment (ncjones) [reply src/api/bank-accounts/bank-account-requests.md:96]:
```suggestion
| code  | String | Authorization code returned returned from third party. |
```

File: src/api/bank-accounts/bank-account-requests.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+| status                   | String             | The current status of the Bank Account Request. Supported values are `created`, `failed` and `done`. |
+| createdAt                | {% dt Timestamp %} | When the Bank Account Request was created.                                                           |
+| createdBy                | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                           |
+| updatedAt                | {% dt Timestamp %} | When the Bank Account Request was updated.                                                           |
+| updatedBy                | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                           |
+| authorizationRedirectUrl | String             | A redirect url to begin the OAuth 2 flow.                                                            |
+
+### Bank Account Request Types
+
+|     Field      |                                   Description                                    |
```
Comment (ncjones) [inline src/api/bank-accounts/bank-account-requests.md:39]:
```suggestion
|     Name      |                                   Description                                    |
```

File: src/api/bank-accounts/bank-account-requests.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+| type                     | String             | The type of Bank Account Request. See [Bank Account Request Types][].                                |
+| status                   | String             | The current status of the Bank Account Request. Supported values are `created`, `failed` and `done`. |
+| createdAt                | {% dt Timestamp %} | When the Bank Account Request was created.                                                           |
+| createdBy                | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                           |
+| updatedAt                | {% dt Timestamp %} | When the Bank Account Request was updated.                                                           |
+| updatedBy                | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                           |
+| authorizationRedirectUrl | String             | A redirect url to begin the OAuth 2 flow.                                                            |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:40]: Perhaps this description could just say: 'A redirect URL to begin the authorization flow.'

File: src/api/bank-accounts/bank-account-requests.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+### Bank Account Request Types
+
+|     Name      |                                   Description                                    |
+| :------------- | :------------------------------------------------------------------------------- |
+| quartz-details | Request will initiate an Oauth 2 flow for authorizing access to account details. |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:46]: Again maybe we should just say authorization flow instead of specifically referencing OAuth 2.

Comment (IsaacMurtagh) [reply src/api/bank-accounts/bank-account-requests.md:46]: Done 👍

File: src/api/bank-accounts/bank-account-requests.md (hunk 5 above)
Comment (IsaacMurtagh) [reply src/api/bank-accounts/bank-account-requests.md:40]: Have change it to something similar.

Comment (MeganSteenkamp) [reply src/api/bank-accounts/bank-account-requests.md:40]: @IsaacMurtagh it looks like it says the same thing?

File: src/api/bank-accounts/bank-account-requests.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+| type                     | String             | The type of Bank Account Request. See [Bank Account Request Types][].                                |
+| status                   | String             | The current status of the Bank Account Request. Supported values are `created`, `failed` and `done`. |
+| createdAt                | {% dt Timestamp %} | When the Bank Account Request was created.                                                           |
+| createdBy                | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                           |
+| updatedAt                | {% dt Timestamp %} | When the Bank Account Request was updated.                                                           |
+| updatedBy                | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                           |
+| authorizationRedirectUrl | String             | A redirect URL to the third-party beginning the authorization flow                                   |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:35]: Missing full stop 😅

File: src/api/bank-accounts/bank-account-requests.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+## Contents
+{:.no_toc .text-delta}
+
+* TOC
+{:toc}
+
+## Models
+
+### Bank Account Request
+{% h4 Mandatory Fields %}
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:23]: I noticed here it says `Mandatory Fields` and below it says `Required Fields` - maybe these should use the same term?

File: src/api/bank-accounts/bank-account-requests.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  type: quartz-details
+  status: created
+  createdAt: 2022-03-31T02:56:29.713Z
+  createdBy: crn:B4u4WZCu3joZFVWT3XjWW3:api-key:MyApiKey
+  updatedAt: 2022-03-31T02:56:29.713Z
+  updatedBy: crn:B4u4WZCu3joZFVWT3XjWW3:api-key:MyApiKey
+  "authorizationRedirectUrl": "https://example.com/authorization-redirect-url"
+{% endjson %}
+
+### Create Bank Account Request **EXPERIMENTAL**
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:80]: The `Authorize` and `Create` titles are around the wrong way.

File: src/api/bank-accounts/bank-account-requests.md (hunk 8 above)
Comment (IsaacMurtagh) [reply src/api/bank-accounts/bank-account-requests.md:23]: Mandory fields is for the model. I'll change required fields to 'fields' as we're doing optional ones with a tag now

File: src/api/bank-accounts/bank-account-requests.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+| type                     | String             | The type of Bank Account Request. See [Bank Account Request Types][].                                |
+| status                   | String             | The current status of the Bank Account Request. Supported values are `created`, `failed` and `done`. |
+| createdAt                | {% dt Timestamp %} | When the Bank Account Request was created.                                                           |
+| createdBy                | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                           |
+| updatedAt                | {% dt Timestamp %} | When the Bank Account Request was updated.                                                           |
+| updatedBy                | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                           |
+| authorizationRedirectUrl | String             | A redirect URL to the third-party beginning the authorization flow.                                  |
```
Comment (mark-scott) [inline src/api/bank-accounts/bank-account-requests.md:35]: Change to `authorizationUrl`

File: src/api/bank-accounts/bank-account-requests.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+{% json %}
+  id: 3KVjuKW2CZCJeJVqPxwkX7
+  accountId: B4u4WZCu3joZFVWT3XjWW3
+  type: quartz-details
+  status: created
+  createdAt: 2022-03-31T02:56:29.713Z
+  createdBy: crn:B4u4WZCu3joZFVWT3XjWW3:api-key:MyApiKey
+  updatedAt: 2022-03-31T02:56:29.713Z
+  updatedBy: crn:B4u4WZCu3joZFVWT3XjWW3:api-key:MyApiKey
+  "authorizationRedirectUrl": "https://example.com/authorization-redirect-url"
```
Comment (mark-scott) [inline src/api/bank-accounts/bank-account-requests.md:78]: Change to `authorizationUrl`

**Review by MeganSteenkamp — DISMISSED** [review]

File: src/api/bank-accounts/bank-account-requests.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+| status           | String             | The current status of the Bank Account Request. Supported values are `created`, `failed` and `done`. |
+| createdAt        | {% dt Timestamp %} | When the Bank Account Request was created.                                                           |
+| createdBy        | {% dt CRN %}       | The User or API Key that created the Bank Account Request.                                           |
+| updatedAt        | {% dt Timestamp %} | When the Bank Account Request was updated.                                                           |
+| updatedBy        | {% dt CRN %}       | The User or API Key that updated the Bank Account Request.                                           |
+| authorizationUrl | String             | A URL to the third-party beginning the authorization flow.                                           |
+
+### Bank Account Request Types
+
+|     Field      |                                      description                                       |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:39]: | Name | Description |

File: src/api/bank-accounts/bank-account-requests.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+      code: 'hKpKupTM391pE10xfQiorMxXarRKAHRhTfH_x',
+    })
+  }
+{% endreqspec %}
+
+{% h4 Fields %}
+
+| Field |  Type  |                                        Description                                         |
+| :---- | :----- | :----------------------------------------------------------------------------------------- |
+| code  | String | Authorization code returned from third-party that can be exchanged for a long lived token. |
```
Comment (MeganSteenkamp) [inline src/api/bank-accounts/bank-account-requests.md:98]: | code  | String | Authorization code returned returned from third party. |

**Review by ncjones — APPROVED** [review]

**Pull request merged**